sdk/resourcemanager/Proto.Client/compute/AvailabilitySetOperations.cs

```diff
@@ -29,7 +29,7 @@ internal AvailabilitySetOperations(GenericResourceOperations genericOperations)
         /// <param name="resourceGroup"> The client parameters to use in these operations. </param>
         /// <param name="availabilitySetName"> The name of the availability set to use. </param>
         internal AvailabilitySetOperations(ResourceGroupOperations resourceGroup, string availabilitySetName)
-            : base(resourceGroup, $"{resourceGroup.Id}/providers/Microsoft.Compute/availabilitySets/{availabilitySetName}")
+            : base(resourceGroup, resourceGroup.Id.AppendProviderResource("Microsoft.Compute", "availabilitySets", availabilitySetName))
```

Comment:
can we use the AvailabilitySetOperations.ResourceType static value here? (same comment for all [Resource]Operations

Reply:
Excellent idea.